javascript_nodejs/17.proactive-messages/dialogs/mainDialog/index.js

```diff
@@ -90,40 +100,60 @@ class MainDialog {
         }
     }
 
-    async completeJob(storage, turnContext, jobIdNumber) {
+    async completeJob(turnContext, jobIdNumber) {
+        // Get the list of jobs.
+        const jobs = await this.jobsList.get(turnContext, {});
+
         // Read from storage
-        let jobInfo = await storage.read([jobIdNumber]);
+        let jobInfo = jobs[jobIdNumber];
 
         // If no job was found, notify the user of this error state.
-        if (isEmpty(jobInfo)){
+        if (isEmpty(jobInfo)) {
             await turnContext.sendActivity(`Sorry no job with ID ${ jobIdNumber }.`);
         } else {
             // Found a job with the ID passed in.
-            const reference = jobInfo[jobIdNumber].reference;
-            const completed = jobInfo[jobIdNumber].completed;
+            const reference = jobInfo.reference;
+            const completed = jobInfo.completed;
 
             // If the job is not yet completed and conversation reference exists,
             // use the adapter to continue the conversation with the job's original creator.
             if (reference && !completed) {
                 // Since we are going to proactively send a message to the user who started the job,
                 // we need to create the turnContext based on the stored reference value.
-                await this.adapter.continueConversation(reference, async (turnContext) => {
+                await this.adapter.continueConversation(reference, async (proactiveTurnContext) => {
```

Comment:
Please change to generate a callback that includes the job number in the proactive message to the original user; otherwise, they don't know which of their jobs just finished.